### PR TITLE
Dynamic notifications

### PR DESCRIPTION
**Pull request by OfirMatasas**
## Description: 
This pull request handles triggering dynamic notifications, according to the current location of the device.

## Type of change:
- [ ] Bug fix
- [x] New feature
- [x] Fix/new infrastructure

## Background context of the changes:
This is essential part of the project.

## The ticket this PR closes:
https://ofirlindekel.atlassian.net/browse/NOTIFY-122?atlOrigin=eyJpIjoiYTBmOTQ1YmU3NzFmNGM5OWJlZmE0YTY1ZTY3OGFhNjMiLCJwIjoiaiJ9

## Scenarios that were tested during this implementation:
* Application is closed
* Application is running
* Several dynamic locations

## Insert screenshots / video of test results (for all relevant platforms)
image

### CODE REVIEW

File: Notify/Notify/Notify/AppShell.xaml.cs [hunk 1]
```diff
@@ -203,30 +224,29 @@ private List<Notification> getAllArrivedLocationNotifications(List<string> desti
                 .FindAll(
                     notification =>
                     {
-                        bool isLocationNotification = notification.Type is NotificationType.Location;
+                        bool isCurrentType = notification.Type is NotificationType.Location || notification.Type is NotificationType.Dynamic;
```
Comment (LinMoskovitch) [inline Notify/Notify/Notify/AppShell.xaml.cs:227]: I am not sure I understood the name of the variable, currentType of what?

File: Notify/Notify/Notify/ViewModels/FriendRequestPageViewModel.cs [hunk 2]
```diff
@@ -83,6 +107,17 @@ private async void PopulateUsersList()
             UsersList = JsonConvert.DeserializeObject<List<Friend>>(usersListJson);
             UsersSelectionList = UsersList = await AzureHttpClient.Instance.GetNotFriendsUsers();
         }
+        
+        private async void PopulateFriendRequestsList()
+        {
+            //string usersListJson = Preferences.Get(Constants.PREFERENCES_FRIEND_REQUESTS, "");
+            
+            //UsersList = JsonConvert.DeserializeObject<List<Friend>>(usersListJson);
```
Comment (LinMoskovitch) [inline Notify/Notify/Notify/ViewModels/FriendRequestPageViewModel.cs:115]: Do we need this?

File: Notify/Notify/Notify/AppShell.xaml.cs (hunk 1 above)
Comment (OfirMatasas) [reply Notify/Notify/Notify/AppShell.xaml.cs:227]: It supposed to be "isCorrectType".
Probably caused by mistyping.
Although I think that "isRelevantType" is a better name after all. Changing it now.

File: Notify/Notify/Notify/ViewModels/FriendRequestPageViewModel.cs (hunk 2 above)
Comment (OfirMatasas) [reply Notify/Notify/Notify/ViewModels/FriendRequestPageViewModel.cs:115]: Not at the moment.
Should be on a different branch. Removing it now.

File: Notify/Notify/Notify/AppShell.xaml.cs (hunk 1 above)
Comment (LinMoskovitch) [reply Notify/Notify/Notify/AppShell.xaml.cs:227]: `isRelevantType` is good

**Review by LinMoskovitch — APPROVED** [review]

**Pull request merged**